actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -283,20 +292,23 @@ class CookieJar #:nodoc:
       # $& => example.local
       DOMAIN_REGEXP = /[^.]*\.([^.]*|..\...|...\...)$/
 
-      def self.build(req, cookies)
-        new(req).tap do |hash|
+      def self.build(req, cookies, cookies_default_options: nil)
+        new(req, cookies_default_options: cookies_default_options).tap do |hash|
           hash.update(cookies)
         end
       end
 
       attr_reader :request
 
-      def initialize(request)
+      def initialize(request, cookies_default_options: nil)
         @set_cookies = {}
         @delete_cookies = {}
         @request = request
         @cookies = {}
         @committed = false
+        @cookies_default_options = cookies_default_options || {}
```

Comment:
I think a better place for this is to actually copy what `cookies_serializer` does and place it in a (renamed) `railties/lib/rails/generators/rails/app/templates/config/initializers/cookies_serializer.rb.tt` initializer. Maybe of like this:

```ruby
# Be sure to restart your server when you modify this file.

Rails.application.config.action_dispatch.tap do |config|
  # Specify a serializer for the signed and encrypted cookie jars.
  # Valid options are :json, :marshal, and :hybrid.
  config.cookies_serializer = :json
  # Specify default options for cookies.
  # Valid options are :path, :domain, :tld_length, :expires, :secure, and :httponly.
  config.default_cookies_options = {}
end
```